[WebAPI] Add 3 test cases for Presentation connection lifecycle

yhe39 opened the pull request:
According the the comment, modify the test cases:
Use testharness assert to check several point in one test cases.
Remove the license information.
Delete the test cases that covered by idlharness.html.

Test reslults: under crosswalk for Android 20.50.532.0.
presentation/PresentationConnection_onclosed-manual.html  2/3 pass
presentation/PresentationConnection_onconnected-manual.html pending (the onconnect is not triggered)
presentation/PresentationConnection_onterminated-manual.html 1/3 pass

yhe39 committed:
[WebAPI] Add 3 test cases for Presentation connection lifecycle

According the the comment, modify the test cases:
Use testharness assert to check several point in one test cases.
Remove the license information.
Delete the test cases that covered by idlharness.html.

Test reslults: under crosswalk for Android 20.50.532.0.
presentation/PresentationConnection_onclosed-manual.html  2/3 pass
presentation/PresentationConnection_onconnected-manual.html pending (the onconnect is not triggered)
presentation/PresentationConnection_onterminated-manual.html 1/3 pass

yhe39 commented:
@Honry @zqzhang PTAL. thanks a lot!

zqzhang commented on presentation/PresentationConnection_onclosed-manual.html:
```diff
@@ -0,0 +1,46 @@
+
+<script>
+  setup({explicit_timeout: true});
+  var startBtn = document.getElementById("presentStartBtn");
```
This var is un-referenced. Then you don't need the `id` of the button.

zqzhang commented on presentation/PresentationConnection_onclosed-manual.html:
```diff
@@ -0,0 +1,46 @@
+  var startBtn = document.getElementById("presentStartBtn");
+  var startPresentation = function () {
+    async_test(function(t) {
+      var url = "display/presentation.html";
```
Prefer a `support/` sub-directory for the presentation page.

zqzhang commented on presentation/PresentationConnection_onclosed-manual.html:
```diff
@@ -0,0 +1,46 @@
+          });
+          connection.close();
+          test(function() {
+            assert_equals(connection.state, "closed");
```
This test is wrong. See line 33. We have to check the closed status after we make sure it is really closed, that is the close event is fired.

zqzhang commented on presentation/PresentationConnection_onclosed-manual.html:
```diff
@@ -0,0 +1,46 @@
+      var request = new PresentationRequest(url);
+      request.start()
+        .then(function(connection) {
+          assert_true(connection instanceof PresentationConnection);
```
Seems this assertion shall be in a test step, or you can make a promise test which wrappers test step in the test itself, see https://github.com/w3c/testharness.js/blob/master/testharness.js#L518

zqzhang commented on presentation/PresentationConnection_onclosed-manual.html:
```diff
@@ -0,0 +1,46 @@
+          }, 'the connection state is closed after close');
+      })
+      .catch(function(ex) {
+        assert_unreached(ex.name + ":" + ex.message);
```
See comment on line 29.

zqzhang commented on presentation/PresentationConnection_onconnected-manual.html:
```diff
@@ -0,0 +1,41 @@
+      request.start()
+        .then(function(connection) {
+          assert_true(connection instanceof PresentationConnection);
+          connection.onconnect = t.step_func_done(function() {
```
If we are sure the `connect` event must be fired, we can test it together with `onclose` in sequence. See `EventWatcher` introduction at

https://github.com/w3c/testharness.js/blob/master/docs/api.md#promise-tests

zqzhang commented on presentation/PresentationConnection_onterminated-manual.html:
```diff
@@ -0,0 +1,51 @@
+      request.start()
+        .then(function(connection) {
+          assert_true(connection instanceof PresentationConnection);
+          connection.onterminate = t.step_func_done(function(evt) {
```
Same comment as for PresentationConnection_onconnected-manual.html

yhe39 committed:
[WebAPI] Modify the test case according to the comments

Remove the un-referenced variable.
Remove the wrong test checkpoin.
Change the presentation.html directory under support
The async_test can't change to promise_test as the event handler
The onconnect is not triggered in this code, if combine with close/terminate, may miss this checkpoint. So design as 3 test cases.

yhe39 committed:
[WebAPI] change the presentation floder name

yhe39 commented:
@Honry @zqzhang PTAL. thanks a lot!

zqzhang commented on presentation/support/presentation.html:
```diff
@@ -0,0 +1,25 @@
+  navigator.receiver.connectionList
+    .then(function(list) {
+      list.onconnectionavailable = function(connections) {
+        addConnection(connections[connections.length-1]);
```
Put spaces before and after `-`. Others look good to me.

By the way, what's your testing result?

yhe39 replied:
test results:
PresentationConnection_onclosed-manual.html   pass 
PresentationConnection_onconnect-manual.html  fail (onconnect is not triggered)
PresentationConnection_onterminate-manual.html    fail (onterminate is not triggered but triggered the onclosed)

yhe39 commented:
Close the PR as mission complete :)
